### PR TITLE
Delete the entire recurring event or only the selected event

### PR DESCRIPTION
**Pull request by Angatupyry**
## What's new

- New field added into ScheduleTask model. `except_dates`. An array that stores the dates that the user does not want a job to run. 
- Added an endpoint to regenerate the schedule based on exception days. 
- Added a pop-up with the option to delete all events or just the selected event. 

[chrome-capture-2023-7-22.webm](https://github.com/open-rmf/rmf-web/assets/37310205/878a158f-a35f-43a8-9394-a292ab16a101)

[chrome-capture-2023-7-22 (1).webm](https://github.com/open-rmf/rmf-web/assets/37310205/8901584e-9d3d-4190-8d31-f8076e251f35)

## Self-checks

- [ ] I have prototyped this new feature (if necessary) on Figma 
- [ ] I'm familiar with and follow this [Typescript guideline](https://basarat.gitbook.io/typescript/styleguide)
- [ ] I added unit-tests for new components
- [ ] I tried testing edge cases
- [ ] I tested the behavior of the components that interact with the backend, with an e2e test

### CODE REVIEW

**Review by aaronchongth — CHANGES_REQUESTED** [review]
Thanks for handling this epic feature! I'll test the feature over the weekend for multiple days to check that scheduling and deletion is working. For now, I've left some comments, let me know what you think!

File: packages/dashboard/src/components/tasks/tasks-app.tsx [hunk 1]
```diff
@@ -12,6 +12,10 @@ import {
   TableContainer,
   Toolbar,
   Tooltip,
+  FormControl,
+  RadioGroup,
+  FormControlLabel,
+  Radio,
```
[the remark this answers is not in the document]
Comment (Angatupyry) [reply packages/dashboard/src/components/tasks/tasks-app.tsx]: I thought our precommit did it automatically. I will keep this in mind for future implementations
Thanks for catching it!
Done cb9f5bfe37cc43cc3a59bd98b3538da5bebdecb3 !

File: packages/dashboard/src/components/tasks/tasks-app.tsx [hunk 2]
```diff
@@ -368,6 +389,40 @@ export const TasksApp = React.memo(
...
+          );
+          return;
+        }
+        if (!rmf) {
+          return;
+        }
+        try {
+          if (scheduleDeleteValue === ScheduleDeleteOptions.CURRENT) {
+            console.log(exceptDateRef.current.toISOString());
+            console.log(exceptDateRef.current);
```
[the remark this answers is not in the document]
Comment (Angatupyry) [reply packages/dashboard/src/components/tasks/tasks-app.tsx]: Done b96ccfea44190891c5c936eafe60c1a29a1aa30f !

File: packages/dashboard/src/components/tasks/tasks-app.tsx [hunk 3]
```diff
@@ -368,6 +389,40 @@ export const TasksApp = React.memo(
...
+        ev.preventDefault();
+        const task = eventsMap.current[Number(currentEventId)];
+        if (!task) {
+          console.error(
+            `Failed to delete scheduled task: unable to find task for event ${currentEventId}`,
+          );
+          return;
+        }
+        if (!rmf) {
+          return;
```
[the remark this answers is not in the document]
Comment (Angatupyry) [reply packages/dashboard/src/components/tasks/tasks-app.tsx]: Sure thing!
I used `throw` in order to capture the error and return it in the catch statement 
c8c43e395caae7ccc858008b7663d30914b2cc62

File: packages/api-server/api_server/routes/tasks/scheduled_tasks.py [hunk 4]
```diff
@@ -129,6 +135,25 @@ async def get_scheduled_task(task_id: int) -> ttm.ScheduledTask:
     return task
 
 
+@router.put("/{task_id}/clear")
+async def regenerate_schedule_by_except_date(
```
[the remark this answers is not in the document]
Comment (Angatupyry) [reply packages/api-server/api_server/routes/tasks/scheduled_tasks.py]: Yeah. Makes sense!
11099e641c38858c1364e96abd00c9502c74c73d

File: packages/api-server/api_server/routes/tasks/scheduled_tasks.py [hunk 5]
```diff
@@ -129,6 +135,25 @@ async def get_scheduled_task(task_id: int) -> ttm.ScheduledTask:
...
+    task_id: int,
+    except_date: datetime,
```
[the remark this answers is not in the document]
Comment (Angatupyry) [reply packages/api-server/api_server/routes/tasks/scheduled_tasks.py]: 11099e641c38858c1364e96abd00c9502c74c73d

**Review by aaronchongth — APPROVED** [review]
Awesome stuff! Thanks! LGTM

**Pull request merged**